File: src/GLFMmatlab/pdf_functions/pdf_n.m
function pdf = pdf_n(x,Zp, Bd, mu, w, s2Y, params)
    %
    % Likelihood function for count data
    % Eq. (8) in the paper:
    % "General Latent Feature Models for Heterogeneous Datasets"
    %
    % Inputs:
    %    x: count-data observation
    %   Bd: K*1  weight vector for a particular dimension d
    %   Zp: 1*K, binary vector of feature assignment,
    %       where K: number of latent features
    %  s2Y: Gaussian noise variance  for u
    % mu,w: Hyper-parameters linked to the transformation described in the paper

    pdf = normcdf(f_n_1(x+1,mu,w), Zp*Bd,sqrt(s2Y) ) - ...
        normcdf(f_n_1(x,mu,w), Zp*Bd, sqrt(s2Y) );
end

File: src/GLFMR/pdf_n.R
#' @param B: K*R matrix
#' @param Zp: 1*K, where K: number of latent features
#' @return the pdf for nominal data
pdf_n<-function(x,Zp, B, mu, w, s2Y, params){
  source("f_p_1.R")
 pdf_out<-pnorm(f_p_1(x+1,mu,w),Zp%*%B,sqrt(s2Y))- pnorm(f_p_1(x,mu,w),Zp%*%B,sqrt(s2Y))
 return(pdf_out)
}

